polaris-react/src/components/ResourceItem/ResourceItem.tsx

```diff
@@ -163,60 +164,63 @@ class BaseResourceItem extends Component<CombinedProps, State> {
     let handleMarkup: React.ReactNode = null;
 
+    const itemPaddingInline: React.ComponentProps<typeof Box>['padding'] =
+      polarisSummerEditions2023 ? '3' : {xs: '4', sm: '5'};
+    const itemPaddingBlock: React.ComponentProps<typeof Box>['padding'] = '3';
+
+    const gapBetweenCheckboxAndMedia: React.ComponentProps<
+      typeof HorizontalStack
+    >['gap'] = polarisSummerEditions2023 ? '3' : '4';
+    const gapBetweenOwnedAndChildren: React.ComponentProps<
+      typeof HorizontalGrid
+    >['gap'] = polarisSummerEditions2023 ? '3' : '5';
+
     if (selectable) {
       const checkboxAccessibilityLabel =
         name || accessibilityLabel || i18n.translate('Polaris.Common.checkbox');
 
       handleMarkup = (
-        <UseFeatures>
-          {({polarisSummerEditions2023}) => (
-            <div onClick={this.handleLargerSelectionArea}>
-              <Bleed marginBlock="2" marginInline="3">
-                <Box
-                  zIndex="var(--pc-resource-item-content-stacking-order)"
-                  paddingInlineStart="3"
-                  paddingInlineEnd="3"
-                  paddingBlockStart={polarisSummerEditions2023 ? '2' : '3'}
-                  paddingBlockEnd="2"
-                >
-                  <div onClick={stopPropagation}>
-                    <div onChange={this.handleLargerSelectionArea}>
-                      <UseId>
-                        {(id) => (
-                          <Checkbox
-                            id={id}
-                            label={checkboxAccessibilityLabel}
-                            labelHidden
-                            checked={selected}
-                            disabled={loading}
-                          />
-                        )}
-                      </UseId>
-                    </div>
-                  </div>
-                </Box>
-              </Bleed>
-            </div>
-          )}
-        </UseFeatures>
+        <div
+          className={styles.CheckboxWrapper}
+          onClick={stopPropagation}
+          onChange={this.handleLargerSelectionArea}
+        >
+          <UseId>
+            {(id) => (
+              <Checkbox
+                id={id}
+                label={checkboxAccessibilityLabel}
+                labelHidden
+                checked={selected}
+                disabled={loading}
+                bleedInlineStart={itemPaddingInline}
+                bleedInlineEnd={
+                  media
+                    ? gapBetweenCheckboxAndMedia
+                    : gapBetweenOwnedAndChildren
+                }
+                bleedBlockStart={itemPaddingBlock}
+                bleedBlockEnd={itemPaddingBlock}
+                fill
+                labelClassName={styles.CheckboxLabel}
+              />
+            )}
+          </UseId>
+        </div>
       );
     }
 
     if (media || selectable) {
       ownedMarkup = (
-        <UseFeatures>
-          {({polarisSummerEditions2023}) => (
```

Comment:
Hoisted the `useFeatures` hook to the wrapping component and passed `features` as a prop to avoid having these wrapping components everywhere, and simplify a bit of the logic for setting the above variables.